### PR TITLE
Meillisearch: module cleanup

### PR DESCRIPTION
**Pull request by happysalada**
###### Motivation for this change

meilisearch module cleanup

- switch noAnalytics default to true since the crate is by default compiled without the analytics feature. (it gave me some problems on compilation when packaging, that is why I haven't included it in the package by default)
- add some missing environment variables. coming from https://docs.meilisearch.com/reference/features/configuration.html#options . The notable exception is the snapshot feature. I left it out since I don't have time to test it. Hopefully the next person can improve upon this.
- The one problem where I haven't found a solution is the masterKey secret. Note that by default this is not used. The default running mode is `development` which doesn't require that key. I haven't found a way around the fact that upstream wants to read the secret from an environment variable. I opened an issue asking them to enable reading it from a file, it got moved into a feature request list.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/search/meilisearch.nix [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  ###### implementation
+
+  config = mkIf cfg.enable {
+    systemd.services.meilisearch = {
+      description = "MeiliSearch daemon";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      environment =
+        let
+          masterKey = mkIf (cfg.masterKeyFile != null) (builtins.readFile cfg.masterKeyFile);
```
Comment (aanderse) [inline nixos/modules/services/search/meilisearch.nix:104]: This is a secrety and is being read into the nix store :disappointed:

Comment (happysalada) [reply nixos/modules/services/search/meilisearch.nix:104]: I also don't like it.
upstream can't read a secret from a file, I've opened an issue upstream for that.
So until they enable this upstream, I do not know how to do this better unfortunately.
I'm happy to wait if you think that is better.

File: nixos/modules/services/search/meilisearch.nix [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+    systemd.services.meilisearch = {
+      description = "MeiliSearch daemon";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      environment =
```
Comment (mohe2015) [inline nixos/modules/services/search/meilisearch.nix:109]: Just for the master key "EnvironmentFile="?

Comment (happysalada) [reply nixos/modules/services/search/meilisearch.nix]: thank you for the review!
Actually it looks like a better idea.
The secret would end in the environment which is not good in terms of security, but at least it doesn't end up in the nixStore.
I've implemented the change. Let me know what you think.

**Comment by happysalada** [comment]
I'm going to merge this for now, if there is additional feedback, I'll be happy to take care of it in a separate PR.

File: nixos/modules/services/search/meilisearch.nix [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    payloadSizeLimit = mkOption {
+      description = ''
+        Sets the maximum size of accepted JSON payloads.
+        Value must be given in bytes or explicitly stating a base unit.
+        For example, the default value can be written as 107374182400, '107.7Gb', or '107374 Mb'.
+        Default is ~ 100 MB
+      '';
+      default = "104857600";
+      type = types.str;
+    };
```
Comment (mweinelt) [inline nixos/modules/services/search/meilisearch.nix:101]: This option is not hooked up to anything.

**Pull request merged**